lib/saluki-components/src/destinations/datadog_status_flare/mod.rs

```diff
@@ -0,0 +1,172 @@
+#[allow(unused)]
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+#[allow(unused)]
+use datadog_protos::agent::{
+    GetFlareFilesRequest, GetFlareFilesResponse, GetStatusDetailsRequest, GetStatusDetailsResponse, RemoteAgent,
+};
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_config::GenericConfiguration;
+use saluki_core::components::{
+    destinations::{Destination, DestinationBuilder, DestinationContext},
+    ComponentContext,
+};
+use saluki_env::helpers::remote_agent::RemoteAgentClient;
+use saluki_error::GenericError;
+use saluki_event::DataType;
+use serde::Deserialize;
+use tracing::debug;
+
+const DEFAULT_ID: &str = "agent-data-plane";
+const DEFAULT_DISPLAY_NAME: &str = "Agent Data Plane";
+const DEFAULT_API_ENDPOINT: &str = "localhost:13580";
+
+/// TODO
+/// wrapped client inside an `Option` to fix Deserialization problems
+#[derive(Deserialize)]
+pub struct DatadogStatusFlareConfiguration {
+    #[serde(default = "default_id", rename = "remote_agent_id")]
+    id: String,
+
+    #[serde(default = "default_display_name", rename = "remote_agent_display_name")]
+    display_name: String,
+
+    #[serde(default = "default_api_endpoint", rename = "remote_agent_api_endpoint")]
+    api_endpoint: String,
```

Comment:
We should make this be `api_port`, since we don't ever want to expose a port on anything other than `localhost`.

The default port value should also be 5102, which puts it sequentially with the other two ADP ports, 5100 (admin API) and 5101 (telemetry API).

Reply:
Question: 
We would use `localhost` in the case that the Agent and ADP runs in the same machine. Would there be instances in which ADP runs on another machine? Maybe containers? 

Mostly, asking to understand the possible deployment topologies 😄

Reply:
For containerized environments, everything is currently on the same machine: standalone Docker, Kubernetes, etc.

If ADP ever became a mechanism for doing remote aggregation (pool of ADP workers running on a different node than the application send metrics/logs/traces/etc) _then_ this might matter... but I can't really think of a situation where ADP will ever really be decoupled from a local Agent.